components/Event/index.js

```diff
@@ -0,0 +1,96 @@
+import { useState } from "react";
+import EventForm from "../EventForm";
+import { useRouter } from "next/router";
+import { StyledListItem } from "./Event.styled";
+import { formatDate } from "../../resources/dateUtils";
+
+export default function Event({ event, kidsData, mutate }) {
```

Comment:
Here, it's called `kidsData` instead of `kidData`. Pick one for both sides.

What's more, you expect the prop `mutate` bu you're not receiving it. See `pages/[id].js` where you probably forgot to pass it down to `KidProfile`.